feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/nominationPools/claimRewards/validations/Declarations.kt

```diff
@@ -16,23 +15,25 @@ typealias NominationPoolsClaimRewardsValidationSystemBuilder =
     ValidationSystemBuilder<NominationPoolsClaimRewardsValidationPayload, NominationPoolsClaimRewardsValidationFailure>
 
 fun ValidationSystem.Companion.nominationPoolsClaimRewards(
-    assetSourceRegistry: AssetSourceRegistry
+    enoughTotalToStayAboveEDValidationFactory: EnoughTotalToStayAboveEDValidationFactory
 ): NominationPoolsClaimRewardsValidationSystem = ValidationSystem {
     enoughToPayFees()
 
-    sufficientCommissionBalanceToStayAboveED(assetSourceRegistry)
+    sufficientCommissionBalanceToStayAboveED(enoughTotalToStayAboveEDValidationFactory)
 
     profitableClaim()
 }
 
 private fun NominationPoolsClaimRewardsValidationSystemBuilder.sufficientCommissionBalanceToStayAboveED(
-    assetSourceRegistry: AssetSourceRegistry
+    enoughTotalToStayAboveEDValidationFactory: EnoughTotalToStayAboveEDValidationFactory
 ) {
-    enoughTotalToStayAboveED(
-        fee = { it.fee },
-        total = { it.asset.total },
-        existentialDeposit = { assetSourceRegistry.existentialDeposit(it.chain, it.chain.utilityAsset) },
-        error = { NominationPoolsClaimRewardsValidationFailure.ToStayAboveED(it.chain.utilityAsset) }
+    validate(
+        enoughTotalToStayAboveEDValidationFactory.create(
+            fee = { it.fee },
+            total = { it.asset.total },
+            chainWithAsset = { ChainWithAsset(it.chain, it.chain.utilityAsset) },
+            error = { NominationPoolsClaimRewardsValidationFailure.ToStayAboveED(it.chain.utilityAsset) }
+        )
```

Comment:
We can remove the need to call `validate` every time if we call it and use `context(ValidationSystemBuilder<P, S>)` inside `enoughTotalToStayAboveEDValidationFactory`